src/tools/tidy/src/known_bug.rs

```diff
@@ -6,8 +6,22 @@ use crate::walk::*;
 
 pub fn check(filepath: &Path, bad: &mut bool) {
     walk(filepath, |path, _is_dir| filter_not_rust(path), &mut |entry, contents| {
-        let file = entry.path();
-        if !contents.lines().any(|line| line.starts_with("//@ known-bug: ")) {
+        let file: &Path = entry.path();
+
+        // files in "auxiliary" do not need to crash by themselves
+        let test_path_segments =
+            file.iter().map(|s| s.to_string_lossy().into()).collect::<Vec<String>>();
+        let test_path_segments_str =
+            test_path_segments.iter().map(|s| s.as_str()).collect::<Vec<&str>>();
+
+        if !matches!(test_path_segments_str[..], [
+            ..,
+            "tests",
+            "crashes",
+            "auxiliary",
+            _aux_file_rs
```

Comment:
Question: I think the aux test file check be simplified to something like

```rs
let is_crashes_aux_file = |p| !p.is_symlink()
    && p.is_dir()
    && p.components().rev().nth(1).map(|c| c.as_os_str()).is_some_and(|d| d == "auxiliary");
```

Collecting into `Vec<String>` in `test_path_segments` would do a bunch of small string allocations, right?